utils/splunk/splunk_helper.py

```diff
@@ -38,6 +40,40 @@ def auth_session(self):
         session_key = response_json["sessionKey"]
         self.requests_session.headers.update({'Authentication': "Splunk %s" % session_key})
 
+    def create_auth_token(self, token):
+        self.log.debug("Creating a new authentication token")
+        token_path = '/services/authorization/tokens?output_mode=json'
+        name = self.instance_config.username
+        audience = self.instance_config.audience
+        expiry_days = self.instance_config.token_expiration_days
+        payload = {'name': name, 'audience': audience, 'expires_on': "+{}d".format(str(expiry_days))}
+        self.requests_session.headers.update({'Authorization': "Bearer %s" % token})
+        response = self._do_post(token_path, payload, self.instance_config.default_request_timeout_seconds)
+        response.raise_for_status()
+        response_json = response.json()
+
+        new_token = response_json.get("entry")[0].get("content").get("token")
+        self.requests_session.headers.update({'Authorization': "Bearer %s" % new_token})
+        return new_token
+
+    def is_token_valid(self, token):
+        current_time = self._current_time()
+        decoded_token = jwt.decode(token, verify=False, algorithm='HS512')
+        expiry_time = decoded_token.get("exp")
+        expiry_date = datetime.datetime.fromtimestamp(expiry_time)
+        days = (expiry_date.date() - current_time.date()).days
+        # as soon as we are close to expiration day, we need to create auth token
+        if days <= 1:
+            return False, days
+        else:
+            self.log.debug("Token is valid.")
+            self.requests_session.headers.update({'Authorization': "Bearer %s" % token})
```

Comment:
Why update the session header when the Token was still valid? `create_auth_token` does this already, right?

Reply:
done.